yt/fields/geometric_fields.py

```diff
@@ -94,9 +94,10 @@ def _zeros(field, data):
     def _ones(field, data):
         """Returns one for all cells"""
         arr = np.ones(data.ires.shape, dtype="float64")
+        tmp = data.apply_units(arr, field.units)
```

Comment:
I find it weird that `tmp` is the return variable while `arr` never leaves the function. Wouldn't it be more natural the other way around ?